x-pack/plugins/security_solution/public/common/hooks/eql/api.ts

```diff
@@ -3,11 +3,84 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { Unit } from '@elastic/datemath';
 
-import { HttpStart } from '../../../../../../../src/core/public';
-import { DETECTION_ENGINE_EQL_VALIDATION_URL } from '../../../../common/constants';
+import { DataPublicPluginStart } from '../../../../../../../src/plugins/data/public';
+import {
+  EqlSearchStrategyRequest,
+  EqlSearchStrategyResponse,
+} from '../../../../../data_enhanced/common';
 import { EqlValidationSchema as EqlValidationRequest } from '../../../../common/detection_engine/schemas/request/eql_validation_schema';
 import { EqlValidationSchema as EqlValidationResponse } from '../../../../common/detection_engine/schemas/response/eql_validation_schema';
+import { InspectResponse } from '../../../types';
+import { ChartData } from '../../components/charts/common';
+import { getEqlAggsData, getSequenceAggs } from './helpers';
+import { HttpStart } from '../../../../../../../src/core/public';
+import { DETECTION_ENGINE_EQL_VALIDATION_URL } from '../../../../common/constants';
+
+interface PreviewApiParams {
+  data: DataPublicPluginStart;
+  interval: Unit;
+  fromTime: string;
+  toTime: string;
+  signal: AbortSignal;
+}
+
+export interface EqlPreviewResponse {
+  data: ChartData[];
+  totalCount: number;
+  lte: string;
+  gte: string;
+  inspect: InspectResponse;
+}
+
+export const getEqlPreview = async ({
+  data,
+  index,
+  interval,
+  query,
+  fromTime,
+  toTime,
+  signal,
+}: PreviewApiParams & EqlValidationRequest): Promise<EqlPreviewResponse> => {
+  try {
+    const response = await data.search
+      .search<EqlSearchStrategyRequest, EqlSearchStrategyResponse>(
+        {
+          params: {
+            // @ts-expect-error allow_no_indices is missing on EqlSearch
+            allow_no_indices: true,
+            index: index.join(),
+            body: {
+              filter: {
+                range: {
+                  '@timestamp': {
+                    gte: toTime,
+                    lte: fromTime,
+                  },
+                },
+              },
+              query,
+              size: 50,
```

Comment:
This was one of the discuss points in the PR description - I picked a number that was at least double the max daily alerts for showing the warning about narrowing your search.